### ISSUE TITLE
No users setup

### ISSUE THREAD

**Issue by flapjackfritz** [vabene1111/recipes#469]
**Is your feature request related to a problem? Please describe.**
There is no security requirement for my use case, so having an authentication system is annoying. Security is provided by having limited access to the network, cannot access the app without being on the local net, which is only available to trusted actors.

**Describe the solution you'd like**
A default user that is automatically used as the universal account, and always logged in when the site is accessed (no authentication required). Probably a flag or something on setup (maybe when the super user is made) that is just permanently granting full authentication to any attempt to access to the site.

**Describe alternatives you've considered**
Just making a default account and password and giving that to everyone.

**Comment by vabene1111** [comment]
you could use reverse proxy authentication. If you have a network setup that handles security it might already be build in.

It is planned to make a user independet public view version but as many function require a user context this is not a simple flag that can be set somewhere.

it will probably be some time before this is implemented

**Comment by flapjackfritz** [comment]
Ahh cool, I'll checkout the reverse proxy authentication, I suppose that's using Aurelia.

The user context thing is why I was thinking it might be easier to have it have a setup option where it just creates a default user with no password, then just assumes all activity is that user. I looked through the allauth docs though and couldn't find any obvious or easy way to do that sort of login flow.

If I were to do it though, it would probably involve changing the authentication helpers/filters/controller to check if a no-auth configuration is set, and if it is, then don't redirect to login (basically turn off all the auth endpoints) and replace the user context with a fixture that has full permissions to everything.

**Comment by vabene1111** [comment]
yes thats basically what needs to be done. But as that is quite a lot of work for something that is cool but not super important it probably has to wait for a few other things to get done first

**Comment by vabene1111** [comment]
decided that this is out of scope for now
